### PR TITLE
add link to API documentation

### PR DESCRIPTION
**Pull request by ltalirz**
So far, optimade.org is very much developer facing.
After the paper is published, the web site will need to point users in
the right direction.

This is a first step towards this - it's not perfect since the "Try it out" buttons won't work.
Still, I think it's the best documentation we currently have.
I'm deleting the link to the Wiki to make space and since it's not used very much.

Mentioning @CasperWA for info

### CODE REVIEW

**Comment by ltalirz** [comment]
> Do you know if it's possible to not create a new tab/window, but instead embed it on the page?

We could iframe it of course but what would be the benefit?

File: _layouts/default.html [hunk 1]
```diff
@@ -34,9 +34,11 @@
             <br>
             <div class="menu">
         	<a href="/"><img src="/assets/images/home.png" height="15px"></a>
+                <a href="https://petstore.swagger.io/?url=https://raw.githubusercontent.com/Materials-Consortia/optimade-python-tools/master/openapi.json" target="_blank">API Documentation
```
Comment (ml-evs) [inline _layouts/default.html]:
```suggest
<a href="https://petstore.swagger.io/?url=https://raw.githubusercontent.com/Materials-Consortia/optimade-python-tools/master/openapi/openapi.json" target="_blank">API Documentation

Comment (ml-evs) [inline _layouts/default.html]:
```suggestion
<a href="https://petstore.swagger.io/?url=https://raw.githubusercontent.com/Materials-Consortia/optimade-python-tools/master/openapi/openapi.json" target="_blank">API Documentation
```

Comment (ml-evs) [reply _layouts/default.html]: Looks like our last PR changed this link :)

Comment (ltalirz) [reply _layouts/default.html]: thanks for catching this!

**Comment by ltalirz** [comment]
> I support this idea, but do we want to do anything about how our docstrings (in rst format as they are mostly directly from the spec) are rendered in swagger?

If that is easily possible, I'm all for it, of course.
Otherwise I would say, even swagger docstrings in rst format is much better than having to read the spec (which is essentially the current alternative for users)

Also mentioning @gmrigna for comment, whom I forgot

**Comment by ml-evs** [comment]
> If that is easily possible, I'm all for it, of course.

It looks like swagger just uses the generated `openapi.json` file, so writing some script that converts all the descriptions from rst to md is possible, but maybe not a worthwhile use of anyone's time!

**Comment by rartino** [comment]
That type of conversion shouldn't be difficult, basically one can just filter the text segments through pandoc, e.g.:
```
cat description.rst | pandoc -f rst -t markdown > description.md
```
Is there already a script that automatically extracts the relevant text segments out of the main rst file, or were these copy+pasted?

**Comment by ltalirz** [comment]
> Is there already a script that automatically extracts the relevant text segments out of the main rst file, or were these copy+pasted?

There is no main rst file - openapi.json is generated automatically from the python code.

Looking through the [`openapi.json` file](https://github.com/Materials-Consortia/optimade-python-tools/blob/master/openapi/openapi.json), I actually see already a lot of markdown markup in there.

In terms of rst, there are some things like `:val:` that I'm not familiar with.
@ml-evs Is there a particular reason for  using RST in the docstrings?
If markdown is good enough, we can also just switch to markdown.

Note that [one can still use sphinx to build documentation from docstrings written in markdown](https://stackoverflow.com/a/56428123/1069467).

**Comment by rartino** [comment]
> There is no main rst file - openapi.json is generated automatically from the python code.

I meant the specification document since @ml-evs said the docstrings are "mostly directly from the spec". It would be very good not to maintain two sources of documentation if we can. So, to have some form of scripting generating the docstrings automatically out of the specification every time we update formulations in the specification/add new properties, etc., would be a good idea.

From what you are saying, I assume no such scripting is in place. If the python code uses actual python docstrings, I assume there is no simple way to pull them from an external source. This means such a script needs to extract docstrings out of the specification and insert them in the right place in the python file. Does that seem doable?

> In terms of rst, there are some things like :val: that I'm not familiar with.
> @ml-evs Is there a particular reason for using RST in the docstrings?
> If markdown is good enough, we could also just switch to using markdown.

As he said, they are taken directly from the specification document. 

The :val: markup explains what type of literal a literal is (e.g., so you can use different formatting when typesetting them). If you run it through pandoc, they are dropped (since markdown does not support that feature).

**Comment by ltalirz** [comment]
> I meant the specification document since @ml-evs said the docstrings are "mostly directly from the spec".

Right - sorry I overlooked this. The docstrings coming directly from the spec seems like a valid reason to use RST here.

> If the python code uses actual python docstrings, I assume there is no simple way to pull them from an external source. This means such a script needs to extract docstrings out of the specification and insert them in the right place in the python file. Does that seem doable?

Even if one would be able to get this to work somehow it would be very fragile ("mostly" is usually not very easy to implement ;-) ). Let's say that I would not want to have to do this myself.

Short of this, it seems to me that the next best solution is indeed to hook into the generation of the openapi.json and to insert an automatic rst-to-md translator in there.

E.g. one could modify the [`fastapi.openapi.utils.get_openapi` function](https://github.com/tiangolo/fastapi/blob/master/fastapi/openapi/utils.py#L271) and then use something like `new_desc = pypandoc.convert_text(description, 'md', format='rst')`.
One caveat is that this requires pandoc - pypandoc [includes pandoc only on windows and macos](https://pypi.org/project/pypandoc/).
A solution could be to do this conversion only when writing the spec to json format, and to add an extra for the pypandoc dependency.

**Comment by ltalirz** [comment]
I've opened issue https://github.com/Materials-Consortia/optimade-python-tools/issues/184 for improving the appearance of the docstrings - in the meanwhile I suggest we go ahead and merge this.

File: _layouts/default.html [hunk 2]
```diff
@@ -33,10 +33,12 @@
             </div>
             <br>
             <div class="menu">
-        	<a href="index"><img src="assets/images/home.png" height="15px"></a>
+        	<a href="index"><img src="/assets/images/home.png" height="15px"></a>
+                <a href="https://petstore.swagger.io/?url=https://raw.githubusercontent.com/Materials-Consortia/optimade-python-tools/master/openapi/openapi.json" target="_blank">API Documentation
```
Comment (CasperWA) [inline _layouts/default.html]:
```suggestion
                <a href="https://petstore.swagger.io/?url=https://petstore.swagger.io/?url=https://raw.githubusercontent.com/Materials-Consortia/OPTiMaDe/v1.0.0-rc.1/schemas/openapi_schema.json" target="_blank">API Documentation
```

Comment (ltalirz) [reply _layouts/default.html]: Thanks @CasperWA , I fully agree.
I wasn't aware that the spec repository also contained an openapi.json.

Comment (CasperWA) [reply _layouts/default.html]: Sure - it's a direct copy of the one created in `optimade-python-tools` at the moment, but we are working on moving the generation either to a separate repo or into the spec repo.

**Comment by ltalirz** [comment]
I don't remember how this "o" got in but we need to remove it

**Pull request merged**